File: sdks/thunkmetrc-java/wrapper/src/main/java/io/github/thunkier/thunkmetrc/wrapper/models/PaginatedResponse.java
package io.github.thunkier.thunkmetrc.wrapper.models;

import java.util.List;
import com.fasterxml.jackson.annotation.JsonProperty;

public class PaginatedResponse<T> {
    @JsonProperty("Total")
    public Integer Total;
    @JsonProperty("PageSize")
    public Integer PageSize;
    @JsonProperty("Current")
    public Integer Current;
    @JsonProperty("Next")
    public Integer Next;
    @JsonProperty("Previous")
    public Integer Previous;
    @JsonProperty("Data")
    public List<T> Data;
}

File: sdks/thunkmetrc-kotlin/wrapper/src/main/kotlin/io/github/thunkier/thunkmetrc/wrapper/models/PaginatedResponse.kt
package io.github.thunkier.thunkmetrc.wrapper.models

import com.fasterxml.jackson.annotation.JsonProperty

data class PaginatedResponse<T>(
    @JsonProperty("Total")
    val Total: Int? = null,
    @JsonProperty("TotalPages")
    val TotalPages: Int? = null,
    @JsonProperty("PageSize")
    val PageSize: Int? = null,
    @JsonProperty("Current")
    val Current: Int? = null,
    @JsonProperty("Next")
    val Next: Int? = null,
    @JsonProperty("Previous")
    val Previous: Int? = null,
    @JsonProperty("Data")
    val Data: List<T>? = null
)
